demo/src/test/java/org/gbif/pipelines/demo/InterpretedCategoryTransformerTest.java

```diff
@@ -0,0 +1,314 @@
+package org.gbif.pipelines.demo;
+
+import org.gbif.api.vocabulary.BasisOfRecord;
+import org.gbif.api.vocabulary.Continent;
+import org.gbif.api.vocabulary.Country;
+import org.gbif.dwca.avro.Event;
+import org.gbif.dwca.avro.ExtendedOccurence;
+import org.gbif.dwca.avro.Location;
+import org.gbif.pipelines.core.functions.interpretation.DayInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.InterpretationException;
+import org.gbif.pipelines.core.functions.interpretation.MonthInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.YearInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueLineageRecord;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.core.functions.transforms.DwCATermIdentifier;
+import org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer;
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.spatialCategory;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.spatialCategoryIssues;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.temporalCategory;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.temporalCategoryIssues;
+
+/**
+ * Testing the RawRecordtoInterpretedCategoryPTransform.
+ * This test can be further extended when the different categories are defined.
+ */
+public class InterpretedCategoryTransformerTest {
+
+  static final String VALID_OCC_ID = "VALID_ID";
```

Comment:
mixed use of constants names: UPER_CASE, lower_case

Reply:
done.